# 01 - Estructura de los datos

Actividad 4a de la guia + Descripcion de los datos (20 pts). Usa la muestra fija de `data/raw/` (ver notebooks/00_setup_datos.ipynb).

In [1]:
import sys
sys.path.append("..")

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

from src import config
from src import data_loading as dl

sns.set_theme(style="whitegrid")


## 1. Metadata de secuencias (`train.csv`)

Una fila por secuencia.

In [2]:
train_df = dl.load_train_index()
print(train_df.shape)
train_df.head()


(67208, 5)


,path,file_id,sequence_id,participant_id,phrase
0,train_landmarks/5414471.parquet,5414471,1816796431,217,3 creekhouse
1,train_landmarks/5414471.parquet,5414471,1816825349,107,scales/kuhaylah
2,train_landmarks/5414471.parquet,5414471,1816909464,1,1383 william lanier
3,train_landmarks/5414471.parquet,5414471,1816967051,63,988 franklin lane
4,train_landmarks/5414471.parquet,5414471,1817123330,89,6920 northeast 661st road


In [3]:
train_df.dtypes

path                str
file_id           int64
sequence_id       int64
participant_id    int64
phrase              str
dtype: object

In [4]:
train_df["participant_id"].nunique()

94

## 2. Frases objetivo (`phrase`)

Longitud en caracteres por secuencia.

In [5]:
train_df["phrase_len"] = train_df["phrase"].str.len()
train_df["phrase_len"].describe()

count    67208.000000
mean        17.804636
std          5.727504
min          1.000000
25%         12.000000
50%         17.000000
75%         22.000000
max         31.000000
Name: phrase_len, dtype: float64

## 3. Estructura de un archivo de landmarks (`.parquet`)

Un archivo de la muestra fija (`config.SAMPLE_LANDMARK_PATHS`): cuadros por secuencia, columnas por tipo de landmark (face, pose, manos).

In [6]:
sample_path = dl.landmark_path(config.SAMPLE_LANDMARK_PATHS[0])
sample_landmarks = dl.load_landmarks(sample_path)
sample_landmarks.shape

(161461, 1630)

In [7]:
for lt in config.LANDMARK_TYPES:
    cols = dl.get_landmark_columns(sample_landmarks, lt, "x")
    print(lt, len(cols))

face 468
left_hand 21
pose 33
right_hand 21


## 4. Resumen para el informe

`train.csv` tiene 67,208 filas (una por secuencia) y 5 columnas: `path` y `phrase` (texto), `file_id`, `sequence_id` y `participant_id` (enteros). Participan 94 personas, con una cantidad de secuencias muy dispareja entre ellas (mediana ~794, min 1, max 1535). La frase objetivo va de 1 a 31 caracteres (mediana 17).

Cada archivo de landmarks trae ~1000 secuencias indexadas por `sequence_id` (no es una columna). Por cuadro hay 1630 columnas: 468 puntos de rostro, 33 de pose y 21 por cada mano, cada uno en x/y/z, mas la columna `frame`.